packages/worker/src/worker.ts

```diff
@@ -1,9 +1,25 @@
 import { AmqpClient, type Logger } from "@amqp-contract/core";
 import type { AmqpConnectionManagerOptions, ConnectionUrl } from "amqp-connection-manager";
-import type { ContractDefinition, InferConsumerNames } from "@amqp-contract/contract";
+import type { Channel, Message } from "amqplib";
+import type { ConsumerDefinition, ContractDefinition, InferConsumerNames } from "@amqp-contract/contract";
 import { Future, Result } from "@swan-io/boxed";
 import { MessageValidationError, TechnicalError } from "./errors.js";
-import type { WorkerInferConsumerHandlers, WorkerInferConsumerInput } from "./types.js";
+import type {
+  WorkerInferConsumerBatchHandler,
+  WorkerInferConsumerHandler,
+  WorkerInferConsumerHandlers,
+  WorkerInferConsumerInput,
+} from "./types.js";
+
+/**
+ * Internal type for consumer options extracted from handler tuples.
+ * Not exported - options are specified inline in the handler tuple types.
+ */
+type ConsumerOptions = {
+  prefetch?: number;
+  batchSize?: number;
+  batchTimeout?: number;
+};
```

Comment:
why do we have to redefine it ? and it is incorrect, we cannot have prefetch together with batchSize and batchTimeout